File: insert_dll.c
#include<stdio.h>
#include<stdlib.h>
typedef struct Node{
    int data;
    struct Node *next;
    struct Node *prev;
}Node;
Node *insert(Node *head,int d){
    Node *newNode=(Node *)calloc(1,sizeof(Node));
    newNode->prev=NULL;
    newNode->data=d;
    newNode->next=head;
    if (head!=NULL){
        head->prev=newNode;
        }
        // head->prev=newNode;
    return newNode;

    }

void display(Node *head){
    if(head==NULL){
        printf("List is empty");
    }
    while(head->next!=NULL){
        printf("%d->",head->data);
        head=head->next;
    }
    printf("%d\n",head->data);
}
int main(){
    Node *head=NULL;
    head=insert(head,4);
    head=insert(head,20);
    head=insert(head,14);
    display(head);
    // printf("hi");
    return 0;
}

File: linked_list/linked_list/insert_dll.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
        self.prev=None
def insert(head,d):
     n=Node(d)
     n.next=head
     n.prev=None
     if head!=None:
         head.prev=n
     return n
def display(head):
     if head==None:
         print("List empty")
     while(head.next!=None):
         print("{}->".format(head.data),end="")
         head=head.next
     print("{}\n".format(head.data))
head = None
head = insert(head, 4)
head = insert(head, 20)
head = insert(head, 14)
display(head) 